src/nimib/renders.nim

```diff
@@ -16,7 +16,19 @@ proc compileNimToJs(doc: var NbDoc, blk: var NbBlock) =
   let (dir, filename, ext) = doc.thisFile.splitFile()
   let nimfile = dir / (filename & "_nbCodeToJs_" & $doc.newId() & ext).RelativeFile
   let jsfile = tempdir / "out.js"
-  writeFile(nimfile, blk.context["transformedCode"].vString)
+  var codeText = blk.context["transformedCode"].vString
+  if blk.context["isOwnFile"].vBool:
+    var bumpGensymString = """
+import macros
+
+macro bumpGensym() =
+  let _ = gensym()
+
+""" 
+    for i in 0 .. doc.newId():
```

Comment:
so this is maybe a bit odd, from what I understand.

you want a way to have an increasing number of bumpGenSym for each ownFile of js that you compile. So, our JsGlobalScript has no bumpGenSym, correct? it would be ok to have 1 bumpGenSym for first ownFile, 2 bumpGenSym for second ownFile, and so on?

if the above is correct, I think that here we are adding a "sort of random" number of gensym (depending on how many docId we have consumed up to that moment) for each ownFile. So the first ownFile might get 7 bumps, the second ownFile will get at least 8 but might get 11, and so on...

What if instead of this we add a jsOwnFilePrelude (or jsOwnFilePrefix) to NbDoc type that during init is initialized only with the macro bumpGenSym, then every time you compile a ownFile you add a single bump. In this way first file would get a single bump, second ownFile would get two bumps,  and so on... in principle it should not change much* but it feels a bit better.

*there is the off chance that having more bugs will prevent collision but if that is the case then we should every time add a big number of bumps (in principle we could add as many lines in code to be conservative...)

Reply:
> you want a way to have an increasing number of bumpGenSym for each ownFile of js that you compile. So, our JsGlobalScript has no bumpGenSym, correct? it would be ok to have 1 bumpGenSym for first ownFile, 2 bumpGenSym for second ownFile, and so on?

Yes, yes and yes.

> if the above is correct, I think that here we are adding a "sort of random" number of gensym (depending on how many docId we have consumed up to that moment) for each ownFile. So the first ownFile might get 7 bumps, the second ownFile will get at least 8 but might get 11, and so on...

Correct. It makes it a bit of a pain to track down bugs, true. Adding a block which doesn't use nbJs but `newId()` could introduce an unfortunate shift indeed. 

> What if instead of this we add a jsOwnFilePrelude (or jsOwnFilePrefix) to NbDoc type that during init is initialized only with the macro bumpGenSym, then every time you compile a ownFile you add a single bump. In this way first file would get a single bump, second ownFile would get two bumps, and so on... in principle it should not change much* but it feels a bit better.

We could absolutely do that, or just have a counter for this. It hit me that we could do `bumpGensym(i)` and do a loop of `i` gensyms inside a single macro call instead. Having 100 calls to a macro (even though it is simple), might have some bad effects of build times, perhaps. So I'd go with a counter for this and moving the loop inside the macro instead.